sessionmanager/sessionmanager.go

```diff
@@ -0,0 +1,65 @@
+package sessionmanager
+
+import (
+	"sync"
+
+	exchange "github.com/ipfs/go-ipfs-exchange-interface"
+)
+
+type SessionManager interface {
```

Comment:
I am not sure of the current opinions of the @ipfs/go-team regarding passing `context.Context`s to all exported functions (at least) for the purpose of tracing/metrics and also cancellation of calls. In cluster we have been taking the approach of all new stuff has it and updating existing code when it is touched.

Reply:
@lanzafame for things that actually need contexts, yes. These methods really don't seem like they need contexts.

Reply:
But tracing requires everything that you want the trace to flow through have a context parameter regardless of whether it is actually required by the function. As horrible and ugly that is, I think the benefits of observability win out in this case, but that is my opinion.

Reply:
I think this isn't the PR for adding tracing. We want to change this as little as possible.